protein_lm/tests/test_coord_check.py

```diff
@@ -0,0 +1,37 @@
+from mup import make_base_shapes, set_base_shapes, make_base_shapes, set_base_shapes, get_shapes, MuAdam, MuSGD, MuAdamW
+from mup.coord_check import get_coord_data, plot_coord_data
+from functools import partial
+import torch
+from protein_lm.modeling import APTConfig, APTLMHeadModel
+
+
+# not sure how to leverage pytest in the context of coordinate checking
+# this is because visual inspection of coordinate checking results is necessary
+# the test will generate coordinate checking results to a test_results directory for now
+
+if __name__ == "__main__":
+    delta_model = APTLMHeadModel(config=APTConfig(n_embd=200, n_layer=8, num_attention_heads=10, n_inner=200, use_mup=True))
+    delta_model.apply(delta_model._init_weights)
```

Comment:
```suggestion
```
I think only the actual model needs to have `_init_weights` applied? I checked, but you should double check too!

Reply:
I had a look into the mup example for transformer and it seems they do not init_weights when mup is active. 
see https://github.com/microsoft/mup/blob/main/examples/Transformer/main.py#L189 (line189 and Line 310 following)  They only call a weights initialization at the end of the definition of their transformer model. 
https://github.com/microsoft/mup/blob/main/examples/Transformer/model.py#L105C9-L105C28 (Line 105).

We also call initialization 
`# Initialize weights and apply final processing
        self.post_init()`
only that our models function is based on the inherited post_init from transformers PreTrainedModel (Line 1243) 
https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_utils.py#L1243C5-L1243C8

If I understood this right,  you are right, and we have to skipp the _init_weights **when using MuP**.